docs/DeveloperGuide.md

```diff
@@ -1,34 +1,429 @@
 
-|Version| As a ... | I want to ... | So that I can ...|
+---------------------------------------------------------------------------------------------
+
+## *2. Design* 
+### 2.1. Architecture
+![img.png](images/architecture.png)
+
+The **Architecture Diagram** above depicts the high-level design of the NUSMaze. You can always refer to this diagram
+to understand how the different components of NUSMaze interact with each other.
+
+The class [`NusMaze`](https://github.com/AY2021S2-CS2113T-T09-2/tp/blob/master/src/main/java/seedu/duke/NusMaze.java) is where the `main()` 
+method belongs and is reponsible for:
+* When the app launches, initialise and connect different components of the NUSMaze in correct sequence.
+* When the app terminates, shut down all the components.
+
+Architecture Components of NUSMaze:
+* [**`UIManager`**](#22-uimanager-component): The user interface of the app
+* [**`Parser`**](#23-parser-component): Processes commands inputted by the user
+* [**`Command`**](#24-command-component): Executes the user command 
+* [**`Router`**](#25-router-component): Searches shortest router
+* [**`Data`**](#26-data-component): Holds the data of the app in memory
+* [**`Storage`**](#27-storage-component): Reads data from and write data to external text files
+
+Explanations on how each component is designed and how it functions are further elaborated in the following 
+chapters of the developer guide.
+
+### 2.2. UIManager Component 
+![img.png](images/ui_design.png)  
+**API**: [UiManager.java](https://github.com/AY2021S2-CS2113T-T09-2/tp/blob/master/src/main/java/seedu/duke/ui/UiManager.java)
+
+The UI of the application is managed by the `UiManager` class as shown by the class diagram above. The individual UI classes for each feature such as `AliasUi`, `DailyRouteUi` and
+`FavouriteUi` extend the `UiManager` class. The UiManager class consists of the methods that are used to display recurrent messages on the *CLI* and also the utilities to get the user's inputs.
+
+The `UiManager` requires the static string variables from the `CommonMessages` class to obtain the commonly used messages that
+such as the divider and input headers.
+
+The individual UI classes contain the methods that are used to get user inputs specific to the needs of the specific feature that
+it is responsible for. For example, when the routing feature is to be executed, the UI will need to prompt the user to obtain 2
+inputs, namely the `from block` and the `to block`. Hence, the `RouterUi` contains the `getRoutingInfo()` method which will prompt
+the user for these two inputs using the utility methods from the UiManager. Methods to get user input are called upon directly from the command classes of the specific feature command.
+
+The `UiManager` component,
+* displays messages in the *CLI*.
+* provides the individual Ui classes with the utilities to obtain user input specific to their needs.
+
+### 2.3. Parser Component 
+![img.png](images/ParserComponent.png)
```

Comment:
Consider standardizing the UML class diagram for parser component with the class diagram in the Ui component